### PR TITLE
Adding trait for GeneratorAwareExtension

### PR DESCRIPTION
**Pull request by Nyholm**
Blocked by #154

### CODE REVIEW

File: src/Faker/Extension/GeneratorAwareExtensionTrait.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+trait GeneratorAwareExtensionTrait
+{
+    /**
+     * @var Generator|null
+     */
+    private $generator = null;
+
+    public function withGenerator(Generator $generator): self
```
Comment (localheinz) [inline src/Faker/Extension/GeneratorAwareExtensionTrait.php:21]:
```suggestion
    /**
     * @return static
     */
    public function withGenerator(Generator $generator): self
```

File: src/Faker/Extension/GeneratorAwareExtensionTrait.php [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+declare(strict_types=1);
+
+namespace Faker\Extension;
+
+use Faker\Generator;
+
+/**
+ * A helper trait to be used with GeneratorAwareExtension.
+ *
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
```
Comment (localheinz) [inline src/Faker/Extension/GeneratorAwareExtensionTrait.php:12]: See #166.

**Comment by localheinz** [comment]
@Nyholm @pimjansen 

I removed an unnecessary property initialization and added a test - do you want to take a look?

**Review by Nyholm — COMMENTED** [review]
Excellent. Thank you

👍

**Review by localheinz — APPROVED** [review]

**Comment by localheinz** [comment]
Thank you, @Nyholm!

**Comment by Nyholm** [comment]
Thank you for improving the PR and for merging

**Pull request merged**